src/main/java/org/myrobotlab/codec/CodecUtils.java

```diff
@@ -139,7 +139,7 @@ public class CodecUtils {
      * without knowing the target type, e.g. if the target
      * is {@link Object}.
      */
-    private static final Class<?> GSON_DEFAULT_OBJECT_TYPE = LinkedTreeMap.class;
+    private static final Class<?> GSON_DEFAULT_OBJECT_TYPE = LinkedHashMap.class;

```

Comment:
downstream unit tests pass .. probably should start ripping gson out

Reply:
> downstream unit tests pass

That's because, as it turns out, there were no tests for the impacted functionality. I was able to write a test that succeeds for Jackson but fails with GSON because of this change. Reverting the change allows GSON to succeed

```java
@Test
  public void testSingleEncodedData() {
    // This test tests whether getDecodedJsonParameters can
    // handle single-encoded data parameters. This would happen
    // when a client mistakenly only single-encodes the data params.
    
    // Manually defining json so the class field isn't added
    String json = "{\"name\":\"testBall\",\"type\":\"basketball\",\"rating\":7}";
    Object decodedBall = CodecUtils.fromJson(json, Object.class);
    Object[] params = new Object[] {decodedBall};
    System.out.println(Arrays.toString(params));
    System.out.println(decodedBall.getClass().toString());
    Object[] decodedParams = MethodCache.getInstance().getDecodedJsonParameters(TestCatcher.class, "catchBall", params);
    System.out.println(decodedParams[0].getClass());


  }

```

Reply:
Does it matter - thanks to your previous prs we're not usin gson any longer on this branch .. right?

Reply:
Not at the moment but if we ever need to change back it could cause issues, I'd like to keep GSON around for a bit longer just to be safe and remove it and supporting code in a future PR